ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java

```diff
@@ -101,6 +101,31 @@ public void updateExtraData(ReactTextView view, Object extraData) {
       ReactAccessibilityDelegate.resetDelegate(
           view, view.isFocusable(), view.getImportantForAccessibility());
     }
+
+    CustomLineHeightSpan[] customLineHeightSpans =
+        spannable.getSpans(0, spannable.length(), CustomLineHeightSpan.class);
+    CustomStyleSpan[] customStyleSpans =
+        spannable.getSpans(0, spannable.length(), CustomStyleSpan.class);
+    if (customLineHeightSpans.length > 0 && customStyleSpans.length > 0) {
+      int highestLineHeight = 0;
+      for (CustomLineHeightSpan span : customLineHeightSpans) {
+        if (highestLineHeight == 0 || span.getLineHeight() > highestLineHeight) {
+          highestLineHeight = span.getLineHeight();
+        }
+      }
+
+      int highestFontSize = 0;
+      if (highestLineHeight != 0) {
+        for (CustomStyleSpan span : customStyleSpans) {
+          if (highestFontSize == 0 || span.getSize() > highestFontSize) {
+            highestFontSize = span.getSize();
+          }
+        }
+        for (CustomStyleSpan span : customStyleSpans) {
+          span.updateSpan(highestLineHeight, highestFontSize);
```

Comment:
Create a new span and call setText instead of updating the existing span https://developer.android.com/develop/ui/views/text-and-emoji/spans#attach-or-detach

Reply:
Does not follow [span#best-practices](https://developer.android.com/develop/ui/views/text-and-emoji/spans#best-practices). Mutates the span after adding it to the text.

Calling [`view.setText(update)`](https://github.com/fabriziobertoglio1987/react-native/blob/224c58a021adf5e4d915c349500e29dc7c935b24/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java#L91) before the span mutation, correctly aligns the text
https://www.icloud.com/iclouddrive/0e3YRsM1KKEs0SXo0nvkEy9OQ#call_set_text_before_mutation

Calling [`view.setText(update)`](https://github.com/fabriziobertoglio1987/react-native/blob/224c58a021adf5e4d915c349500e29dc7c935b24/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java#L91) after the span mutation, changes the baseline of the parent text component and does not align correctly the child text
https://www.icloud.com/iclouddrive/087kw0jNMTAFnT5pJ-mPHXzaQ#call_set_text_after_mutation